airflow/models/dag.py

```diff
@@ -302,16 +304,20 @@ def __init__(
         self.fileloc = back.f_code.co_filename if back else ""
         self.task_dict: Dict[str, BaseOperator] = {}
 
-        # set timezone from start_date
+        # set timezone from schedule_timezone or start_date
+        if schedule_timezone:
+            self.schedule_timezone = self._get_tzname_or_offset(schedule_timezone)
```

Comment:
Should we do any checks to make sure that the start_date is in the right timezone?

(Just trying to think about potential foot-guns that might surprise users.)

Reply:
+1 on checking `start_date` is either naive (and will be converted to `schedule_timezone`) or matches `schedule_timezone`. Maybe `end_date` as well, although we don’t currently do? (Which is still a footgun.)

Reply:
I don't think we really want to check that the `start_date` time zone matches the `schedule_timezone`.  That's why I wanted to introduce `schedule_timezone` to avoid having to rely on inferring things from `start_date.tzinfo`.

There is no problem to specify the `start_date` in one timezone, the `end_date` in another timezone, and have yet another timezone for the `schedule_timezone`.  I can see people (myself included) using  iso 8601 string like `2021-06-01T12:00:00+01:00` for `start_date` and `schedule_timezone='Europe/Stockholm'`

Reply:
But anyway, I'm removing `schedule_timezone` from the PR, right? Because that changes the public api of the DAG and that requires an AIP, right?

Reply:
> There is no problem to specify the `start_date` in one timezone, the `end_date` in another timezone, and have yet another timezone for the `schedule_timezone`. I can see people (myself included) using iso 8601 string like `2021-06-01T12:00:00+01:00` for `start_date` and `schedule_timezone='Europe/Stockholm'`

Hmm, that makes sense. And since if `schedule_timezone` is _not_ set, `start_date.tzinfo` will be used anyway, there is no behavioural change for people not (yet) modifying their DAGs. And when `schedule_timezone` is set, the user should know what they’re doing.

> But anyway, I'm removing `schedule_timezone` from the PR, right? Because that changes the public api of the DAG and that requires an AIP, right?

I think so, yes, but we’re going to need to have this discussion when explicit per-DAG timezone setup becomes a thing (whether via `schedule_timezone` or another mean) anyway.

Reply:
Ok, removed `schedule_timezone` from DAG constructor constructor